integration/sql/impl/src/visitor.rs

```diff
@@ -690,12 +716,49 @@ impl Visit for Table {
 
 // --- Utils ---
 
-fn get_table_name_from_table_factor(table: &TableFactor) -> Result<Vec<Ident>> {
-    if let TableFactor::Table { name, .. } = table {
-        Ok(name.clone().0)
+fn get_table_name_from_table_factor(
+    table: &TableFactor,
+    context: &Context,
+) -> Result<Option<Vec<Ident>>> {
```

Comment:
Either Result or Option - both do not make sense, unless you want to express some three-value logic, and there are better tools for that like Enums

Reply:
Updated.